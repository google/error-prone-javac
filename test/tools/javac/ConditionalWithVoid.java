/*
 * @test /nodynamiccopyright/
 * @bug 4974927
 * @summary The compiler was allowing void types in its parsing of conditional expressions.
 * @author tball
 *
 * @compile/fail/ref=ConditionalWithVoid.out -XDrawDiagnostics ConditionalWithVoid.java
 */
public class ConditionalWithVoid {
    public int test(Object o) {
        // Should fail to compile since Object.wait() has a void return type.
        System.out.println(o instanceof String ? o.hashCode() : o.wait());
    }
}
