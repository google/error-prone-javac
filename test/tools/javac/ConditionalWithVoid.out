ConditionalWithVoid.java:12:48: compiler.err.neither.conditional.subtype: java.lang.Integer, void
1 error
